crates/astria-conductor/src/data_availability.rs

```diff
@@ -0,0 +1,382 @@
+                    self.get_latest_height = None;
+                    self.get_sequencer_datas(res);
+                }
+
+                Some((height, res)) = self.get_sequencer_datas.join_next(), if !self.get_sequencer_datas.is_empty() => {
+                    self.process_sequencer_datas(height, res);
+                }
+
+                Some((height, res)) = self.process_sequencer_datas.join_next(), if !self.process_sequencer_datas.is_empty() => {
+                    let span = tracing::info_span!("send_sequencer_subsets", height);
+                    span.in_scope(|| self.send_sequencer_subsets(res))
+                        .wrap_err("failed sending sequencer subsets to executor")?;
+                }
+            )
+        }
+        Ok(())
+    }
+
+    fn get_latest_height(&mut self) {
+        let client = self.celestia_client.inner().clone();
+        self.get_latest_height = Some(tokio::spawn(async move {
+            Ok(client.header_network_head().await?.height())
+        }))
+    }
+
+    fn get_sequencer_datas(&mut self, latest_height_res: Result<eyre::Result<u64>, JoinError>) {
+        let latest_height = match latest_height_res {
+            Err(e) => {
+                warn!(error.message = %e, error.cause = ?e, "task querying celestia for latest height failed");
+                return;
+            }
+
+            Ok(Err(e)) => {
+                warn!(error.message = %e, error.cause = ?e, "task querying celestia for latest height returned with an error");
+                return;
+            }
+
+            Ok(Ok(height)) => height,
+        };
+
+        if latest_height <= self.current_block_height {
+            info!(
+                height.celestia = latest_height,
+                height.previous = self.current_block_height,
+                "no new celestia height; not spawning tasks to fetch sequencer blocks"
+            );
+            return;
+        }
+        let first_new_height = self.current_block_height + 1;
+        info!(
+            height.start = first_new_height,
+            height.end = self.current_block_height,
+            "spawning tasks to fetch sequencer blocks for different celestia heights",
+        );
+        for height in first_new_height..=self.current_block_height {
+            let client = self.celestia_client.clone();
+            if self.get_sequencer_datas.contains_key(&height) {
+                warn!(
+                    height,
+                    "getting sequencer data from celestia already in flight, not spawning"
+                );
+            } else {
+                self.get_sequencer_datas.spawn(height, async move {
+                    client.get_sequencer_namespace_data(height).await
+                });
+            }
+        }
+    }
+
+    #[instrument(skip(self, sequencer_data_res))]
+    fn process_sequencer_datas(
+        &mut self,
+        height: u64,
```

Comment:
```suggestion
        da_height: u64,
```
or add comment

Reply:
I have renamed the functions. I believe we need not explicitly talk about `da_height` given that we are in the data_availability module (so that the height is explicitly referring to the DA).